### ISSUE TITLE
scatter_add: Mixing of 0-dim and 1-dim tensors

### ISSUE THREAD

**Issue by manuelcandales** [pytorch/pytorch#110084]
### 🐛 Describe the bug

scatter_add currently allows 0-dim and 1-dim tensors to be mixed together

`
import torch
op = torch.ops.aten.scatter_add.default

op(torch.zeros([1]), 0, torch.zeros([], dtype=torch.long), torch.ones([2]))
op(torch.zeros([]), 0, torch.zeros([2], dtype=torch.long), torch.ones([3]))

torch.zeros([1]).scatter_add_(0, torch.zeros([], dtype=torch.long), torch.ones([2]))
torch.zeros([]).scatter_add_(0, torch.zeros([2], dtype=torch.long), torch.ones([3]))
`

However, documentation explicitly says that all three tensors should have the same number of dimensions
https://pytorch.org/docs/stable/generated/torch.Tensor.scatter_add_.html

Code comments and error messages also state that the tensors should have the same dimensionality.

The behavior is caused by the use of `ensure_nonempty_dim` in line 73 here:
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/ScatterGatherChecks.h#L66-L74
Notice code comment (line 66) and error message (line 74) stating number of dimensions must be the same

Is this behavior desired? If so, then we need to update code comment/error message/documentation.

### Versions

.
